interfaces/tests/sync_int/sync_int.c

```diff
@@ -0,0 +1,211 @@
+//Copyright(c) Microsoft.All rights reserved.
+//Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+
+#include "testrunnerswitcher.h"
+#include "interlocked.h"
+#include "threadapi.h"
+#include "sync.h"
+#include "timer.h"
+
+#include "umock_c/umock_c.h"
+#include "umock_c/umock_c_prod.h"
+
+TEST_DEFINE_ENUM_TYPE(THREADAPI_RESULT, THREADAPI_RESULT_VALUES)
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+static int increment_on_odd_values(void* address)
+{
+    volatile_atomic int32_t* ptr = (volatile_atomic int32_t*)address;
+    while (interlocked_add(ptr, 0) < 98)
+    {
+        int value = interlocked_add(ptr, 0);
+        while (value % 2 != 1)
+        {
+            wait_on_address(ptr, &value, UINT32_MAX);
+            value = interlocked_add(ptr, 0);
+        }
+        interlocked_increment(ptr);
```

Comment:
>interlocked_increment [](start = 8, length = 21)

`(void)interlocked_increment(ptr);`
(to be specific that return value is ignored).

Reply:
Here and everywhere the return value is not used

---
In reply to: [443793257](https://github.com/Azure/azure-c-pal/pull/6#discussion_r443793257) [](ancestors = 443793257)